### DISCUSSION TITLE
Update the Edit Product page to support drafts and new draft-based live preview

### DISCUSSION THREAD

**Discussion by Anubarak** [craftcms/commerce#2358] [Ideas] [answered]
That's basically it, please add different preview targets as they exist for sections for products as well, so they can be previewed in headless mode.

**Comment by lukeholder** [comment]
Update the Edit Product page to support drafts and new draft-based live preview

**Comment by RitterKnightCreative** [comment]
@lukeholder looks like this didn't make it to the 3.1 milestone, perhaps 3.2?

**Comment by RitterKnightCreative** [comment]
BTW: this was marked as a duplicate as early as #291 as well.

**Comment by jsunsawyer** [comment]
We just ran into a weird issue where having revision history would have been super beneficial to troubleshooting things. Chiming in in hopes that this issue gets some love. :)

Also, looks like anything related to product revisions is marked as a duplicate, eventually pointing to this issue, even though it doesn't specifically mention revisions.

**Comment by huelabs** [comment]
Agree with @jsunsawyer. We have thousands of products and this is such a pain point for us. Who edited a product and what did they change? It's anyone's guess without revision history.

**Comment by CMeldgaard** [comment]
Is there a timeline for this feature ?

**Comment by edoardolunardi** [comment]
Hey, we're also using Craft Commerce in headless mode (with a Next.js frontend) and it would be nice to be able to customise the preview target for products, like any other entry 🙏

**Comment by Nellyaa** [comment]
Yes, please! I was surprised that this wasn't added already actually when I just realized :)

**Comment by RitterKnightCreative** [comment]
Now that Craft 4 has universal element editing and revision support, this should be easier to implement, @brandonkelly @lukeholder ?

**Comment by edoardolunardi** [comment]
It would be a nice feature and my team is looking forward to it. Any updates on this? @brandonkelly @lukeholder

**Reply by lukeholder** [reply]
Hey all, this is high on our list but needed to wait for some content modelling and UI changes happening in 5.0 to effectively implement this for products and variants.

Stay tuned to this discussion for updates. Thanks.

**Reply by edoardolunardi** [reply]
Hey Luke, thanks for the reply! Any rough ETA? As I said, my team needs this, and having an ETA will help us to make decisions. Thank you!

**Reply by lukeholder** [reply]
5.0 wont be out till later in 2023

**Comment by lukeholder** [comment, marked as answer]
Craft Commerce 5.0 released earlier this year supports product drafts, and works the save features on the edit screen as entries.
